uploader/upload.go

```diff
@@ -168,25 +177,45 @@ func (u *Uploader) Upload(c *fasthttp.RequestCtx) {
 	}
 	id, bt := u.fetchOwnerAndBearerToken(c)
 
-	obj := object.New()
+	var obj object.Object
 	obj.SetContainerID(*idCnr)
 	obj.SetOwnerID(id)
 	obj.SetAttributes(attributes...)
 
-	var prm pool.PrmObjectPut
-	prm.SetHeader(*obj)
-	prm.SetPayload(file)
-
+	var prm client.PrmObjectPutInit
 	if bt != nil {
-		prm.UseBearer(*bt)
+		prm.WithBearerToken(*bt)
 	}
 
-	if idObj, err = u.pool.PutObject(u.appCtx, prm); err != nil {
-		log.Error("could not store file in neofs", zap.Error(err))
-		response.Error(c, "could not store file in neofs: "+err.Error(), fasthttp.StatusBadRequest)
+	writer, err := u.pool.ObjectPutInit(u.appCtx, obj, u.signer, prm)
+	if err != nil {
+		log.Error("writer init", zap.Error(err))
+		response.Error(c, "writer init: "+err.Error(), fasthttp.StatusInternalServerError)
 		return
 	}
 
+	for {
+		_, err = io.CopyN(writer, file, u.settings.maxObjectSize.Load())
+		if err == nil {
+			continue
+		}
+
+		if !errors.Is(err, io.EOF) {
```

Comment:
`CopyN` doesn't return `EOF`
```suggestion
		if err != nil {
```

Reply:
[Returns](https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/io/io.go;l=371) and without this check, it doesn't work